src/set.js

```diff
@@ -193,39 +207,55 @@ Sk.builtin.set.prototype["issubset"] = new Sk.builtin.func(function (self, other
 });
 
 Sk.builtin.set.prototype["issuperset"] = new Sk.builtin.func(function (self, other) {
+    Sk.builtin.pyCheckArgs("issuperset", arguments, 2, 2);
     return Sk.builtin.set.prototype["issubset"].func_code(other, self);
 });
 
 Sk.builtin.set.prototype["union"] = new Sk.builtin.func(function (self) {
-    var i;
-    var S = new Sk.builtin.set(self);
-    for (i = 1; i < arguments.length; i++) {
-        Sk.builtin.set.prototype["update"].func_code(S, arguments[i]);
-    }
+    var S, new_args;
+
+    Sk.builtin.pyCheckArgs("union", arguments, 1);
+
+    S = Sk.builtin.set.prototype["copy"].func_code(self),
+      new_args = Array.prototype.slice.call(arguments); //copy array
```

Comment:
Generally worth avoiding `slice.call(arguments)` where possible - it kills the optimiser. (Annoying, because it's so much nicer than the manual walk method). That said, I'm not the performance nut here, so if you're OK with it, I am ;)